lona/app.py

```diff
@@ -72,40 +79,73 @@ def resolve_path(self, path):
         )
 
     # decorator ###############################################################
-    def route(self, *args, **kwargs):
-        def decorator(view_class):
+    def route(
+            self,
+            # 1 = lona.MATCH_ALL  https://github.com/python/mypy/issues/10026
+            raw_pattern: Union[str, Literal[1]],
+            name: str = '',
+            interactive: bool = True,
+            http_pass_through: bool = False,
+            frontend_view: Union[None, str, LonaView] = None,
+    ) -> Callable[[Type[LonaView]], None]:
+        def decorator(view_class: Type[LonaView]) -> None:
             self.routes.append(
                 Route(
-                    *args,
+                    raw_pattern=raw_pattern,
                     view=view_class,
-                    **kwargs,
+                    name=name,
+                    interactive=interactive,
+                    http_pass_through=http_pass_through,
+                    frontend_view=frontend_view,
                 )
             )
 
         return decorator
 
-    def middleware(self, middleware_class=None):
-        def decorator(middleware_class):
+    @overload
+    def middleware(self) -> Callable[[Type], None]: ...
+
+    @overload
+    def middleware(self, middleware_class: Type) -> None: ...
+
+    def middleware(self, middleware_class: Optional[Type] = None) -> \
```

Comment:
In cases like this i would rather add ``# NOQA: $ERROR_CODE``

Reply:
Does your comment relates to overloading or to multiline declaration?

Reply:
@maratori Multiline declaration. I don't like the ``\`` syntax

Reply:
Ok. I'll rebase this after #75 is merged.

Reply:
I've decided to not wait for #75.